templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+def log(message)
+  puts message
+  `logger "[PuppetCA CSR Verify] #{message}"`
+end
+
+csr = STDIN.read
+
+settings_file    = "<%= scope.lookupvar("foreman_proxy::etc") %>/foreman-proxy/settings.yml"
```

Comment:
You can do this without needing a template, see https://github.com/theforeman/puppet-foreman/blob/b937165001aac788fbafef0422984a59edaeba87/files/external_node_v2.rb#L13-L17.